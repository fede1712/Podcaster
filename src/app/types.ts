export interface Podcast {
  "im:name": { label: string };
  "im:image": image[];
  summary: { label: string };
  "im:price": { label: string; attributes: { amount: string; currency: string } };
  "im:contentType": { attributes: { term: string; label: string } };
  rights: { label: string };
  title: { label: string };
  link: { attributes: { rel: string; type: string; href: string } };
  id: { label: string; attributes: { "im:id": string } };
  "im:artist": { label: string; attributes: { href: string } };
  category: { attributes: { "im:id": string; term: string; scheme: string; label: string } };
  "im:releaseDate": { label: string; attributes: { label: string } };
}

export type image = { label: string; attributes: { height: string } };

export interface Episode {
  country: string;
  artworkUrl160: string;
  episodeFileExtension: string;
  episodeContentType: string;
  artworkUrl600: string;
  collectionViewUrl: string;
  trackTimeMillis: number;
  shortDescription: string;
  closedCaptioning: string;
  collectionId: number;
  collectionName: string;
  genres: genres[];
  episodeGuid: string;
  description: string;
  releaseDate: string;
  previewUrl: string;
  trackId: number;
  trackName: string;
  feedUrl: string;
  artistIds: [];
  episodeUrl: string;
  artworkUrl60: string;
  contentAdvisoryRating: string;
  trackViewUrl: string;
  kind: string;
  wrapperType: string;
  genreIds: string[];
  primaryGenreName: string;
  currency: string;
  trackCount: number;
  trackExplicitness: string;
  artistName: string;
  collectionCensoredName: string;
  trackCensoredName: string;
  artworkUrl30: string;
  artworkUrl100: string;
  collectionPrice: number;
  trackPrice: number;
  collectionHdPrice: number;
  collectionExplicitness: string;
}

export type genres = { name: string; id: string };
