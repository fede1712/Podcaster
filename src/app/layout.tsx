import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "./globals.css";
import Link from "next/link";
import { Header } from "./ui/header/header";

export const metadata: Metadata = {
  title: "Podcaster",
  description: "Itunes top podcasts",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className="m-8">
        <Header />
        {children}
      </body>
    </html>
  );
}
