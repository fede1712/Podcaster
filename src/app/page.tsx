import { getPodcasts } from "./lib/data";
import { PodcastCard } from "./ui/podcastCard/podcastCard";

export default async function Home() {
  const data = await getPodcasts();
  console.log(data.feed.entry[1]);

  return (
    <div className="grid grid-cols-4 gap-14 m-14">
      {data.feed.entry.map((podcast: any) => {
        return <PodcastCard author={podcast["im:artist"].label} title={podcast["im:name"].label} src={podcast["im:image"][2].label} />;
      })}
    </div>
  );
}
