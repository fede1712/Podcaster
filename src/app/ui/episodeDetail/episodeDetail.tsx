import { Episode } from "@/app/types";

export const EpisodeDetail = ({ actualEpisode }: { actualEpisode: Episode }) => {
  return (
    <div className="p-10 border shadow-xl">
      <h2 className="text-2xl font-bold">{actualEpisode.trackName}</h2>
      <p className="italic py-4 border-b-2">{actualEpisode.description}</p>
      <audio controls className="my-4 w-full">
        <source src={actualEpisode.episodeUrl} type="audio/mp3" />
      </audio>
    </div>
  );
};
