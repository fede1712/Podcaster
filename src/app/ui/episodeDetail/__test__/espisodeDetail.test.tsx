import React from "react";
import { render, screen } from "@testing-library/react";
import { Episode } from "@/app/types";
import { EpisodeDetail } from "../episodeDetail";

test("Renders episode details correctly", () => {
  const episode: Episode = {
    trackName: "Test Episode",
    description: "This is a test episode description.",
    episodeUrl: "http://example.com/test-episode.mp3",
    artworkUrl600: "http://example.com/artwork.jpg",
    trackTimeMillis: 123456,
    artistName: "Test artist name",
    collectionName: "Test collection name",
    releaseDate: "Test realease date",
    trackId: 123456,
  };

  render(<EpisodeDetail actualEpisode={episode} />);
  const titleElement = screen.queryByText("Test Episode");
  expect(titleElement).not.toBeNull();

  const descriptionElement = screen.queryByText("This is a test episode description.");
  expect(descriptionElement).not.toBeNull();

  const { asFragment } = render(<EpisodeDetail actualEpisode={episode} />);

  expect(asFragment()).toMatchSnapshot();
});
