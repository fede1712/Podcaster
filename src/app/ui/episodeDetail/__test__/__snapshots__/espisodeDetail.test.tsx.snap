// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders episode details correctly 1`] = `
<DocumentFragment>
  <div
    class="p-10 border shadow-xl"
  >
    <h2
      class="text-2xl font-bold"
    >
      Test Episode
    </h2>
    <p
      class="italic py-4 border-b-2"
    >
      This is a test episode description.
    </p>
    <audio
      class="my-4 w-full"
      controls=""
    >
      <source
        src="http://example.com/test-episode.mp3"
        type="audio/mp3"
      />
    </audio>
  </div>
</DocumentFragment>
`;
